### ISSUE TITLE
`lines-around-comment`: afterBlockComment: false doesn't report an error

### ISSUE THREAD

**Issue by esthezia** [eslint-stylistic/eslint-stylistic#989]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

With the rule

```
"stylistic/lines-around-comment": ["error", {
    "beforeBlockComment": true,
    "afterBlockComment": false
}]
```

configured, I would expect the following code to throw an error due to the inserted line after the block comment, but it doesn't:

```
var myObject = {};

/**
 * My comment
 */

(function () {
    // function code
})();
```

Instead, I get no error whether `afterBlockComment` is `true` _or_ `false`.

I'm not sure if this is a bug, or the way it was intended to work -- if so, please let me know --, but this is how I would expect it to work, as it seems the most logical.

Thank you!

### Reproduction

<none>

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by 9romise** [comment]
The `afterBlockComment` option in `lines-around-comment` only works when set to true, which is a behavior inherited from upstream. I agree it’s a bit confusing.

And let’s keep this issue open and explore a better approach to achieve what you’re looking for.

**Comment by esthezia** [comment]
What I'm looking for here is that block comments that immediately precede a function/method declaration, or a class property or member declaration, should not be separated from these with a newline, because they clearly refer to that function or property, and so I want them together.

That being said, block comments at the beginning of the file, or that do not precede a function or property declaration, will be allowed to be followed by a newline.
